stdlib/template/autotools.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3.6
+# -*- coding: utf-8 -*-
+"""Provides an exhaustive template that downloads, builds and wraps a software based on ``autoconf`` and ``make``.
+"""
+
+import os
+import nbuild
+import stdlib
+import stdlib.fetch
+import stdlib.extract
+import stdlib.patch
+
+from stdlib.template.configure import configure
+from stdlib.template.make import make
+
+
+def build(
+    build_folder='.',
+    fetch=stdlib.fetch.fetch,
+    extract=stdlib.extract.flat_extract_all,
+    patch=stdlib.patch.patch_all,
+    configure=configure,
+    compile=make,
+    check=lambda: make('check', fail_ok=True),
+    install=lambda: make('install', f'DESTDIR={stdlib.build.current_build().install_cache}'),
+    split=None,
+    deplinker=None,
+):
+    """Downloads, build and wraps a library based on ``autoconf`` and ``make``.
+
+    This exhaustive template is made of 9 steps:
+        * ``fetch``
+        * ``extract``
+        * ``patch``
+        * ``configure``
+        * ``compile``
+        * ``check``
+        * ``install``
+        * ``split``
+        * ``dependency linking``
+
+    For each one of these steps, a function is called. This template simply calls all of them in the above order.
+    All of those functions can be given as arguments, but each one of them have a default value that is explained below.
+    If any of these function is ``None``, the step is skipped.
+
+    **Fetch**
+
+        This step is used to download the source code. The default value is :py:func:`.fetch` with no argument.
+
+    **Extract**
+
+        This step is used to extract the downloaded source code. The default value is :py:func:`.flat_extract_all` with no argument.
+
+    **Patch**
+
+        This step is used to patch the downloaded source code. The default value is :py:func:`.patch_all` with no argument.
+
+    From now one, the current working directory is changed in favor of ``build_folder`` (which defaults to ``.``).
```

Comment:
If you meant to say 'starting from now', it should be written that way:
```suggestion
    From now on, the current working directory is changed in favor of ``build_folder`` (which defaults to ``.``).
```